src/Markup/Avalonia.Markup.Xaml/Data/Binding.cs

```diff
@@ -195,6 +262,12 @@ private static PathInfo ParsePath(string path)
             return result;
         }
 
+        private static Type LookupAncestorType(string ancestorTypeName)
+        {
+            //TODO: What is our syntax for type lookup here?
+            throw new NotImplementedException();
```

Comment:
Would simply `$parent[ns:ControlType]` work here? To do this I think we'd need to make `Binding` a `MarkupExtension` so we have access to the XAML context. This is how it is in WPF but we couldn't do it previously because of how OmniXAML worked(https://github.com/OmniGUI/OmniXAML/issues/74) but we should be able to do it now we're using Portable.Xaml.